src/cdk/coercion/number-property.ts

```diff
@@ -10,8 +10,16 @@
 export function coerceNumberProperty(value: any): number;
 export function coerceNumberProperty<D>(value: any, fallback: D): number | D;
 export function coerceNumberProperty(value: any, fallbackValue = 0) {
+  return isNumberValue(value) ? Number(value) : fallbackValue;
+}
+
+/**
+ * Whether the provided value is considered a number.
+ * @docs-private
+ */
+export function isNumberValue(value: any): boolean {
```

Comment:
It would be better to prefix with an `_` to make it completely clear that this is not a public API (rather than being public and undocumented)